src/services/ContributingDeveloperAuditService/ContributingDeveloperAuditService.ts

```diff
@@ -0,0 +1,42 @@
+import { SOOS_CONSTANTS } from "../../constants";
+import { ScmType } from "../../enums";
+import { soosLogger } from "../../logging";
+import GitHubAudit from "./scm/GitHub/GitHubAudit";
+import { ContributingDeveloper } from "./scm/GitHub/api";
+import FileSystem from "fs";
+import * as Path from "path";
+
+export interface IContributingDeveloperAudit {
+  audit(implementationParams: Record<string, string>): Promise<ContributingDeveloper[]>;
+}
+
+class ContributingDeveloperAuditService {
+  private scmImplementation: IContributingDeveloperAudit;
+
+  constructor(scmType: ScmType) {
+    if (scmType === ScmType.GitHub) {
+      this.scmImplementation = new GitHubAudit();
+    } else {
+      throw new Error("Unsupported SCM type");
+    }
+  }
+
+  public async audit(implementationParams: Record<string, string>) {
+    return this.scmImplementation.audit(implementationParams);
```

Comment:
validate the params here, e.g.
```suggestion
    this.auditProvider.validate(implementationParams);
    return this.auditProvider.audit(implementationParams);
```